src/xGadgetron/pGadgetron/tests/test2.py

```diff
@@ -30,19 +30,19 @@ def test_main(rec=False, verb=False, throw=True):
     data_path = examples_data_path('MR')
     AcquisitionData.set_storage_scheme('memory')
     input_data = AcquisitionData(
-            data_path + '/simulated_MR_2D_cartesian_Grappa2.h5')
-    test.check(input_data.norm())
+            data_path + '/autogenerated_simulated_MR_2D_cartesian_Grappa2.h5')
```

Comment:
I guess this is why you install them into `data_path`? But wouldn't it have the same problem if not being able to update these files if read-only?

Reply:
Yes, good point, that's why they gotta be in the data_path directory... But I'm sorry, I don't understand the 2nd part of the comment...

Reply:
that's because I meant "of not being able...". If a sys-admin installed SIRF somewhere (e.g. for our STFC cloud), a user wouldn't be able to update the installed files, i.e. they would be read-only. Of course, a user probably wouldn't run the tests, so this is probably irrelevant. I still find it a bit weird to install and then modify test files.

But this might be too hard, in which case I'd be happy to let it go, and you could raise an issue to sort it out later (this has been going on for long enough 😉 )

Reply:
We ensured that during `build` the output of the Shepp Logan simulation goes to the ${CMAKE_CURRENT_BINARY} and only writes to the $SHARED_DIR location using the `install` command.
As far as I can tell, we are installing the data from the `$SIRF_PATH/data/` as well.
All the Python tests and C++ tests have either `${CURRENT_BINARY_DIR}` or `$SIRF_SOURCE_DIR` as working directories and don't write to the install path, and don't modify the files.

So shouldn't people be able to run the tests independent of their access rights?

Reply:
In the TrajectoryPreparation we do
```
std::stringstream hdr_stream;
serialize(hdr, hdr_stream);

AcquisitionsInfo ai(hdr_stream.str());
mr_acq.set_acquisitions_info(ai);
```
This is all done in memory, nonetheless, the call to serialize will cause the error due to the version compliations. As long as you never call serialize, it will not detect any version problem. That's why it never turned up before.

Reply:
I see now that this call to `serialize` is in memory (i.e. it writes to a `stringstream`). So, I guess I'm wrong then and the file doesn't get modified. Sorry! Let's go ahead then and just clean the last bits up.